synphot/observation.py

```diff
@@ -694,3 +695,218 @@ def as_spectrum(self, binned=True, wavelengths=None):
+
+    References
+    ----------
+    .. [1] Howell, S. B. 2000, *Handbook of CCD Astronomy* (Cambridge, UK:
+        Cambridge University Press)
+    .. [2] Merline, W. & Howell, S. B. *A Realistic Model for Point-sources
+        Imaged on Array Detectors: The Model and Initial Results*. ExA, 6:163
+        (1995)
+
+    Returns
+    -------
+    sn : `~astropy.units.Quantity`
+        The signal to noise ratio of the given observation in units of
+        sqrt(electrons).
+    """
+    readnoise = _get_shotnoise(readnoise)
+    gain_err = _get_shotnoise(gain * ad_err)
+
+    pixel_terms = npix * (1 + npix / n_background)
+    detector_noise = (background + darkcurrent +
+                      readnoise ** 2 + gain_err ** 2)
+
+    sn = counts / np.sqrt(counts + pixel_terms * detector_noise)
+
+    return sn
+
+
+@u.quantity_input(snr=np.sqrt(1 * u.electron),
+                  countrate=u.electron / u.s,
+                  npix=u.pixel,
+                  n_background=u.pixel,
+                  background_rate=u.electron / u.pixel / u.s,
+                  darkcurrent_rate=u.electron / u.pixel / u.s,
+                  readnoise=u.electron / u.pixel,
+                  gain=u.electron / u.adu,
+                  ad_err=u.adu / u.pixel)
+def exptime_from_howell_snr(snr, countrate,
+                            npix=1 * u.pixel,
+                            n_background=np.inf * u.pixel,
+                            background_rate=0 * u.electron / u.pixel / u.s,
+                            darkcurrent_rate=0 * u.electron / u.pixel / u.s,
+                            readnoise=0 * u.electron / u.pixel,
+                            gain=1 * u.electron / u.adu,
+                            ad_err=ad_err_default):
+    """
+    Returns the exposure time needed in units of seconds to achieve
+    the specified (idealized theoretical) signal to noise ratio
+    (from pg 57-58 of [1]_).
+
+    Parameters
+    ----------
+    snr : `~astropy.units.Quantity`
+        The signal to noise ratio of the given observation in units of
+        sqrt(electrons).
+    countrate : `~astropy.units.Quantity`
+        The counts per second with units of electron/second.
+    npix : `~astropy.units.Quantity`, optional
+        Number of pixels under consideration for the signal with units of
+        pixels. Default is 1 * astropy.units.pixel.
+    n_background : `~astropy.units.Quantity`, optional
+        Number of pixels used in the background estimation with units of
+        pixels. Default is set to np.inf * astropy.units.pixel
+        such that there is no contribution of error due to background
+        estimation. This assumes that n_background will be >> npix.
+    background_rate : `~astropy.units.Quantity`, optional
+        Photons per pixel per second due to the backround/sky with units of
+        electrons/second/pixel.
+        Default is 0 * (astropy.units.electron /
+                        astropy.units.second / astropy.units.pixel)
```

Comment:
No need for so much indentation in a docstring.
```suggestion
        astropy.units.second / astropy.units.pixel)
```